scipy/spatial/transform/_rigid_transformation.pyx

```diff
@@ -0,0 +1,1978 @@
+              case `composition` contains the result of composing each
+              transformation in the other object with the single
+              transformation.
+            - Both ``p`` and ``q`` contain ``N`` transformations. In this case
+              each transformation ``p[i]`` is composed with the corresponding
+              transformation ``q[i]`` and `output` contains ``N``
+              transformations.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as Tf
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+
+        Compose two transformations:
+
+        >>> tf1 = Tf.from_translation([1, 0, 0])
+        >>> tf2 = Tf.from_translation([0, 1, 0])
+        >>> tf = tf1 * tf2
+        >>> tf.translation
+        array([1., 1., 0.])
+        >>> tf.single
+        True
+
+        When applied to a vector, the composition of two transformations is
+        applied in right-to-left order.
+
+        >>> t1, r1 = [1, 2, 3], R.from_euler('z', 60, degrees=True)
+        >>> t2, r2 = [0, 1, 0], R.from_euler('x', 30, degrees=True)
+        >>> tf1 = Tf.from_components(t1, r1)
+        >>> tf2 = Tf.from_components(t2, r2)
+        >>> tf = tf1 * tf2
+        >>> tf.apply([1, 0, 0])
+        array([0.6339746, 3.3660254, 3.       ])
+        >>> tf1.apply(tf2.apply([1, 0, 0]))
+        array([0.6339746, 3.3660254, 3.       ])
+
+        When at least one of the transformations is not single, the result is a
+        stack of transformations.
+
+        >>> tf1 = Tf.from_translation([1, 0, 0])
+        >>> tf2 = Tf.from_translation([[0, 2, 0], [0, 0, 3]])
+        >>> tf = tf1 * tf2
+        >>> tf.translation
+        array([[1., 2., 0.],
+               [1., 0., 3.]])
+        >>> tf.single
+        False
+        >>> len(tf)
+        2
+        """
+        len_self = len(self._matrix)
+        len_other = len(other._matrix)
+        if not(len_self == 1 or len_other == 1 or len_self == len_other):
+            raise ValueError("Expected equal number of transformations in both "
+                             f"or a single transformation in either object, "
+                             f"got {len_self} transformations in first and "
+                             f"{len_other} transformations in second object.")
+
+        result = np.matmul(self._matrix, other._matrix)
+        if self._single and other._single:
+            result = result[0]
+        return self.__class__(result, copy=False)
+
+    @cython.embedsignature(True)
+    def __pow__(RigidTransformation self, int n):
+        """Compose this transformation with itself `n` times.
+
+        If `n` is negative, the inverse of the transformation is composed with
+        itself `n` times. In other words,
+        ``p ** -abs(n) == p.inv() ** abs(n)``.
+
+        Non-integer values of `n` are not currently supported.
```

Comment:
Am I right that general power can be implemented using multiplication of exponential coordinates (similar to multiplication of rotation vector)?

Reply:
Yes, that's correct. This is how screw linear interpolation works.

Reply:
I expect we'll land on ScLERP for non-integer powers, but want to have a separate discussion on it once this is in.

Reply:
Considering how straightforward it to implement, might be a good idea to introduce "ScLERP" and then maybe reconsider or add other options.

Reply:
I think that makes sense, but let's defer to a later PR

Reply:
OK, it's your call, but the implementation will be one-liner.